integration/tools/validation/src/main/java/alluxio/cli/UfsVersionValidationTask.java

```diff
@@ -0,0 +1,74 @@
+package alluxio.cli;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.underfs.UnderFileSystemFactoryRegistry;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This validates the UFS configuration for a users configured UFS path. It ensures that the
+ * version of the UFS library is available (if configured).
+ */
+@ApplicableUfsType(ApplicableUfsType.Type.ALL)
+public class UfsVersionValidationTask extends AbstractValidationTask {
+
+  private final String mUfsPath;
+  private final AlluxioConfiguration mConf;
+
+    /**
+     * Create a new instance of {@link UfsVersionValidationTask}.
+     *
+     * @param ufsPath the UFS URI to test
+     * @param ufsConf the configuration for the UFS URI
+     */
+  public UfsVersionValidationTask(String ufsPath, AlluxioConfiguration ufsConf) {
+    mUfsPath = ufsPath;
+    mConf = ufsConf;
+  }
+
+  @Override
+  protected ValidationTaskResult validateImpl(Map<String, String> optionMap) {
+    UnderFileSystemConfiguration ufsConf =
+          UnderFileSystemConfiguration.defaults(mConf).createMountSpecificConf(optionMap);
+    String configuredVersion = mConf.get(PropertyKey.UNDERFS_VERSION);
+    List<String> availableVersions =
+        UnderFileSystemFactoryRegistry.getSupportedVersions(mUfsPath, ufsConf);
+    ValidationTaskResult result = new ValidationTaskResult();
+    result.setName(getName());
+    result.setDesc("Validates that the configured UFS version exists as a library on the "
+        + "system.");
+
+    if (!mConf.isSetByUser(PropertyKey.UNDERFS_VERSION)) {
+      result.setState(ValidationUtils.State.SKIPPED);
+      result.setOutput("The UFS version was not configured by the user.");
```

Comment:
Do you think it's possible there's a case where the user should have configured the `UNDERFS_VERSION` but did not?

Reply:
Users should only configure UFS version in the case that UFSes support a version. In our current implementation, Alluxio doesn't require a version to be specified, even if there is more than one version supported. It just picks a default if it is not configured. So, it is not necessary for a ever configure a version for a UFS to be chosen.

However, there is a a chance that the default version used is _not_ compatible with the particular UFS. This may induce runtime errors upon connection or data transfer. In those cases we defer to the other UFS checks to flesh our those errors. This check is designed just to make sure that the configured version corresponds to a valid Alluxio UFS library. There's no good way to detect is the UFS is _actually_ compatible at runtime. And even if it's not, any errors that occur likely won't say anything along the line of "hey, you're using the wrong client version". It will be much more difficult to decipher.

tl;dr yes, it is possible but determining the case in which the UFS version is wrong is a non-trivial problem

Reply:
I see, that's fair to say. (Should tldr be on the top?)